addons/website_event/models/event_question.py

```diff
@@ -0,0 +1,97 @@
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+
+
+class EventQuestion(models.Model):
+    _name = 'event.question'
+    _rec_name = 'title'
+    _order = 'sequence,id'
+    _description = 'Event Question'
+
+    title = fields.Char(required=True, translate=True, compute='_compute_title', readonly=False, store=True)
+    question_type = fields.Selection([
+        ('simple_choice', 'Selection'),
+        ('text_box', 'Text Input'),
+        ('name', 'Name'),
+        ('email', 'Email'),
+        ('phone', 'Phone'),
+        ('company', 'Company'),
+    ], default='simple_choice', string="Question Type", required=True)
+    event_type_id = fields.Many2one('event.type', 'Event Type', ondelete='cascade')
+    event_id = fields.Many2one('event.event', 'Event', ondelete='cascade')
+    answer_ids = fields.One2many('event.question.answer', 'question_id', "Answers", copy=True)
+    sequence = fields.Integer(default=10)
+    once_per_order = fields.Boolean('Ask once per order',
+                                    help="If True, this question will be asked only once and its value will be propagated to every attendees."
+                                         "If not it will be asked for every attendee of a reservation.")
+    is_mandatory_answer = fields.Boolean('Mandatory Answer')
+    is_identification_question = fields.Boolean('Identification Question', compute='_compute_is_identification_question')
+
+    @api.constrains('event_type_id', 'event_id')
+    def _constrains_event(self):
+        if any(question.event_type_id and question.event_id for question in self):
+            raise UserError(_('Question cannot belong to both the event category and itself.'))
+
+    @api.depends('question_type')
+    def _compute_title(self):
+        for question in self:
+            question_type_dict = dict(self._fields['question_type'].selection)
+            if question.question_type not in ['simple_choice', 'text_box']:
+                question.title = question_type_dict.get(question.question_type)
+            elif question.title in question_type_dict.values():
+                question.title = False
+
+    @api.depends('question_type')
+    def _compute_is_identification_question(self):
+        for question in self:
+            question.is_identification_question = question.question_type in ['name', 'email', 'phone', 'company']
+
+    def write(self, vals):
+        """ We add a check to prevent changing the question_type of a question that already has answers.
+        Indeed, it would mess up the event.registration.answer (answer type not matching the question type). """
+
+        if 'question_type' in vals:
+            questions_new_type = self.filtered(lambda question: question.question_type != vals['question_type'])
+            if questions_new_type:
+                answer_count = self.env['event.registration.answer'].search_count([('question_id', 'in', questions_new_type.ids)])
```

Comment:
I know this is the original code, but can't we use 'answer_ids' field instead of doing a search_count ?